modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/util/OpenAPIUtil.java

```diff
@@ -51,6 +52,20 @@ public static String escapeVersion(OpenAPIYAML openAPIYAML) {
 	}
 
 	public static String formatSingular(String s) {
+		Set<String> wordsExceptionsToKeepFormat = new HashSet<>();
+
+		wordsExceptionsToKeepFormat.add("address");
```

Comment:
It's makes sense to me. The only thing is that it could create more breaking changes. For example if someone created a custom endpoint ending in `business` and is already integrated to their internal system, this change would break it. Because the actual endpoint ends in `busines` and this change would update it to end in `business.` What do you think?

Reply:
Then just hardcode address:

```
if (s.endsWith("address")) {
    return s;
}
```